src/main/java/org/jenkinsci/plugins/GithubAuthorizationStrategy.java

```diff
@@ -102,7 +102,7 @@ public ACL getACL(@Nonnull AbstractItem item) {
 
     @Nonnull
     public ACL getACL(@Nonnull Job<?,?> job) {
-        if(job instanceof WorkflowJob && job.getProperty(BranchJobProperty.class) != null || job instanceof AbstractProject) {
+        if(job instanceof WorkflowJob || job instanceof AbstractProject) {
```

Comment:
So why check types at all?

Reply:
I don't know enough about Jenkins internals to be able to answer that question (and I'm not the original author), it's just that restricting to jobs with `BranchJobProperty` is too restrictive.
Is it possible that not all `Job` are `AbstractProject`?

Reply:
In practice `Job`s are

* `AbstractProject`—freestyle, matrix, Maven, the traditional types
* `WorkflowJob`
* some obscure other cases, like external monitoring jobs

Generally, you should not downcast without a good reason.